TransformationSystem/DB/TransformationDB.py

```diff
@@ -1525,3 +1521,94 @@ def __addDirectory( self, path, force ):
       else:
         successful.append( lfn )
     return {"OK":True, "Value": len( res['Value']['Successful'] ), "Successful":successful, "Failed": failed }
+
+  def setMetadata( self, path, usermetadatadict ):
+    """ It can be applied to a file or to a directory (path). For a file, add the file to Transformations if the updated metadata dictionary passes the filter.
+        For a directory, add the files contained in the directory to the Transformations if the the updated metadata dictionary passes the filter.
+    """
+    gLogger.info( "setMetadata: Attempting to set metadata %s to %s" % (usermetadatadict, path) )
+    successful = {}
+    failed = {}
+    if isinstance( path, dict ):
```

Comment:
I found why I had introduced this test... I had some type mismatch problem, because I was using:
  `@checkCatalogArguments`
  with 
 ` def setMetadata( self, path, metadatadict ):`
in Resources/Catalog/TSCatalogClient

So you are right, the test is not necessary.